packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,240 @@
+<bar>["]                                  {this.begin("bar_title");}
+<bar_title>[^"]+                          {return 'BAR_TITLE';}
+<bar_title>["]" "*                        {this.popState(); this.begin("bar_data");}
+<bar_data>"["" "*                         {this.begin('bar_data_entries');}
+<bar_data_without_label,bar_data_entries>(?:[+-]?\d+(?:\.\d+)?)+(?:" "*[,]" "*(?:[+-]?\d+(?:\.\d+)?)+)*" "*    {return 'BAR_DATA'}
+<bar_data_entries>"]"" "*                 {this.popState(); this.popState(); this.popState()}
+<bar_data_without_label>"]"" "*           {this.popState(); this.popState()}
+<bar>[^\s\[]+" "*                           {this.begin("bar_data"); return 'BAR_TITLE';}
+<bar>"["" "*                              {this.begin('bar_data_without_label');}
+
+
+
+
+["][`]                                    { this.begin("md_string");}
+<md_string>[^`"]+                         { return "MD_STR";}
+<md_string>[`]["]                         { this.popState();}
+["]                                       this.begin("string");
+<string>["]                               this.popState();
+<string>[^"]*                             return "STR";
+
+[A-Za-z]+                                 return 'ALPHA';
+":"                                       return 'COLON';
+\+                                        return 'PLUS';
+","                                       return 'COMMA';
+"="                                       return 'EQUALS';
+\=                                        return 'EQUALS';
+"*"                                       return 'MULT';
+\#                                        return 'BRKT';
+[\_]                                      return 'UNDERSCORE';
+"."                                       return 'DOT';
+"&"                                       return 'AMP';
+\-                                        return 'MINUS';
+[0-9]+                                    return 'NUM';
+\s                                        return 'SPACE';
+";"                                       return 'SEMI';
+[!"#$%&'*+,-.`?\\_/]                      return 'PUNCTUATION';
+<<EOF>>                                   return 'EOF';
+
+/lex
+
+%start start
+
+%% /* language grammar */
+
+start
+  : eol start
+  | SPACE start
+  | directive start
+  | XYCHART chartConfig CHART_CONFIG_END document
+	| XYCHART CHART_CONFIG_END document
+	;
+
+chartConfig
+  : chart_orientation {yy.setOrientation($1.trim());}
+  ;
+
+document
+	: /* empty */
+	| document line
+	;
+
+line
+	: statement eol
+	;
+
+statement
+  :
+  | SPACE statement
+	| directive
+	| title title_value  { $$=$2.trim();yy.setDiagramTitle($$); }
+  | X_AXIS parseXAxis
+  | Y_AXIS parseYAxis
+  | parseLine
+  | parseBar
+	;
+
+parseLine
+  : LINE LINE_DATA {yy.setLineData('', $2.split(',').map(d => Number(d.trim())));}
+  | LINE LINE_TITLE LINE_DATA {yy.setLineData($2.trim(), $3.split(',').map(d => Number(d.trim())));}
```

Comment:
If a `LINE_DATA` was not a token, but a non-termial symbol, you would not need to split them by comma, you would get them parsed